### PR TITLE
Add span logs

### PR DESCRIPTION
**Pull request by ruiaraujo**
If an object matching the opentracing span interface is passed to the request, log the events of the tcp connection.

### CODE REVIEW

**Comment by ruiaraujo** [comment]
:+1:

File: lib/request.ts [hunk 1]
```diff
@@ -45,6 +66,10 @@ export interface ServiceClientRequestOptions extends RequestOptions {
    * and there is no activity on that socket
    */
   readTimeout?: number;
+  /**
+   * Opentracing like span interface to log events
+   */
+  span?: Span;
```
Comment (oporkka) [inline lib/request.ts]: We could/should document this.

**Comment by oporkka** [comment]
:+1:

**Comment by addityasingh** [comment]
👍

**Pull request merged**